packages/vike-react-apollo/src/withFallback.tsx

```diff
@@ -0,0 +1,122 @@
+import React, { ComponentType, ReactNode, Suspense } from 'react'
+import { ErrorBoundary, FallbackProps } from 'react-error-boundary'
+import { ApolloConsumer } from '@apollo/client/index.js'
+type RetryOptions = { retryQuery?: boolean }
+type RetryFn = (options?: RetryOptions) => void
+
+type ErrorFallbackProps = {
+  error: { message: string } & Record<string, unknown>
+  retry: RetryFn
+}
+
+type Loading<T> = ComponentType<T> | ReactNode
+type Error<T> = ComponentType<T & ErrorFallbackProps> | ReactNode
+
+type WithFallbackOptions<T> = {
+  Loading?: Loading<T>
+  Error?: Error<T>
+}
+
+export function withFallback<T extends object = Record<string, never>>(
+  Component: ComponentType<T>,
+  options?: WithFallbackOptions<T>
+): ComponentType<T>
+export function withFallback<T extends object = Record<string, never>>(
```

Comment:
I feel like the third overload isn't worth it? I feel like it adds more noise than adding benefits?

Reply:
There is only 1 overload, I don't understand.

Reply:
Ah, sorry, you're right.